components/storage/blockdevice/COMPONENT_SDIO/SDIOBlockDevice.cpp

```diff
@@ -0,0 +1,397 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2019 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#include "sdio_api.h"
+#if DEVICE_SDIO_ASYNC
+#include "us_ticker_api.h"
+#endif
+#include "SDIOBlockDevice.h"
+#include "platform/mbed_debug.h"
+
+using namespace mbed;
+
+#if DEVICE_SDIO
+
+/*
+ *  defines
+ */
+
+#define SDIO_DBG 0       /*!< 1 - Enable debugging */
+#define SDIO_CMD_TRACE 0 /*!< 1 - Enable SD command tracing */
+
+#define SDIO_BLOCK_DEVICE_ERROR_WOULD_BLOCK -5001           /*!< operation would block */
```

Comment:
I would make this enum rather than macros. SImilar to types like `qspif_bd_error` (an example how error code look like).